kubernetes/migration/versions.go

```diff
@@ -45,6 +45,13 @@ func renamePlugin(p *corefile.Plugin, to string) (*corefile.Plugin, error) {
 	return p, nil
 }
 
+func removeArgument(o *corefile.Option, remove string) (*corefile.Option, error) {
```

Comment:
Reading more closely, I see that this function actually removes the entire option, not just the argument.
So, `removeOptionIfHasArgument` would be more indicative of what this does. And it only checks the first argument... so ... `removeOptionIfFirstArgumentIs`
_However_, now that I better understand what the function is doing, I'm not sure how re-usable this behavior would actually be.  Might be better to just to do it as a one off as you did original, but with less generic name, e.g.

```
var proxyRemoveHttpsGoogleProtocol = func(o *corefile.Option) (*corefile.Option, error) {
    if o.Args[0] == "https_google" {
      return nil, nil
    }
    return o, nil
  }
```

Also, the function would fail if `o.Args` is zero length.  This may not be allowed in `protocol`, but cant hurt to  put a check in there just to be safe, e.g. `if len(o.Args) > 0 && o.Args[0] == "https_google"`

Reply:
Yes, that makes sense. I reverted it back to the original fn with the additional check for length.